### ISSUE TITLE
Awaitable singleton raises RuntimeError

### ISSUE THREAD

**Issue by thiromi** [meadsteve/lagom#243]
Hey there 👋 

I'm using this interesting library in one of my projects and I faced a situation where I wanted awaitable singletons. Using the tests from this repo, I was able to reproduce the behaviour I wanted:

```python
@dataclass
class MyComplexDep:
    some_number: int

    @classmethod
    async def asyc_loader(cls):
        return cls(10)


@pytest.mark.asyncio
async def test_async_singleton_do_not_raise_runtime_error(container: Container):
    container[Awaitable[MyComplexDep]] = Singleton(MyComplexDep.asyc_loader)  # type: ignore

    assert (await container[Awaitable[MyComplexDep]]) == MyComplexDep(some_number=10)  # type: ignore
    assert (await container[Awaitable[MyComplexDep]]) == MyComplexDep(some_number=10)  # type: ignore
```

this piece of code raises 

```
RuntimeError: cannot reuse already awaited coroutine
```

Because of https://github.com/python/cpython/issues/70075 where is not possible to await a coroutine multiple times.

I have some code that might enable this feature, I'm just not sure if other async implementations like tornado, twisted, gevent is going to support this change.

**Comment by thiromi** [comment]
Fixed in https://github.com/meadsteve/lagom/pull/244

**Comment by meadsteve** [comment]
and now released as 2.4.2
